cpp/bench/CMakeLists.txt

```diff
@@ -32,6 +32,7 @@ function(ConfigureBench)
     PRIVATE raft::raft
             $<$<BOOL:${ConfigureBench_DIST}>:raft::distance>
             $<$<BOOL:${ConfigureBench_NN}>:raft::nn>
+            GTest::gtest
```

Comment:
The benchmark shouldn't depend on gtest. I think, we should try to break the offending dependency instead (which is not so trivial).

Reply:
...Or maybe we could live with the gtest dependency in benchmarks while we solve the problem with the header dependencies? Any thoughts, @cjnolet ?

Reply:
100% agreed w/ @achirkin here. I was going to type up a similar comment but he beat me to it!

Reply:
> The benchmark shouldn't depend on gtest. I think, we should try to break the offending dependency instead (which is not so trivial).

My previous comment was in response to this, btw. I don't think we should be crossing the gtest and benchmark dependencies. We should try and pull out the shared bits into a separate header (and maybe try to find a good place in the `include/raft/`?) for the shared bits but keep gtest-specific things in test and benchmark-specific things in bench.

Reply:
I'm also okay opening a github issue about this and keeping the gtest dependency in here for now but adding a comment here that references the issue.

Reply:
Opened #1153